crates/y-sweet-core/src/doc_connection.rs

```diff
@@ -160,12 +172,24 @@ impl DocConnection {
                     protocol.handle_sync_step1(&awareness, sv)
                 }
                 SyncMessage::SyncStep2(update) => {
-                    let mut awareness = a.write().unwrap();
-                    protocol.handle_sync_step2(&mut awareness, Update::decode_v1(&update)?)
+                    if can_write {
+                        let mut awareness = a.write().unwrap();
+                        protocol.handle_sync_step2(&mut awareness, Update::decode_v1(&update)?)
+                    } else {
+                        Err(sync::Error::PermissionDenied {
+                            reason: "Token does not have write access".to_string(),
+                        })
+                    }
                 }
                 SyncMessage::Update(update) => {
-                    let mut awareness = a.write().unwrap();
-                    protocol.handle_update(&mut awareness, Update::decode_v1(&update)?)
+                    if can_write {
+                        let mut awareness = a.write().unwrap();
+                        protocol.handle_update(&mut awareness, Update::decode_v1(&update)?)
+                    } else {
+                        Err(sync::Error::PermissionDenied {
+                            reason: "Token does not have write access".to_string(),
+                        })
```

Comment:
Do we plan to make the clients aware of the authorization level as well? Not sure how these errors are logged/captured/etc but as of now this is an expected code path, since clients will send updates as though they have full write access.